api/src/test/java/org/openmrs/validator/PersonValidatorTest.java

```diff
@@ -231,5 +234,28 @@ public void validate_shouldNotFailWhenDeathCauseNotCodedIsSet() {
 
 		Assert.assertFalse(errors.hasErrors());
 	}
+	
+	@Test()
+	public void validate_shouldNotFailWhenOneAddressIsNull() {
+		Person person = new Patient(1);
+		Set<PersonAddress> addresses =  new HashSet<>();
+		addresses.add(null);
+		
+		person.setDead(true);
+		person.setCauseOfDeathNonCoded("Some text describing Cause of Death");
+		person.setPersonVoided(true);
+		person.setPersonVoidReason("voidReason");
+		person.setGender("g");
+		person.setAddresses(addresses);
+		
+		Errors errors = new BindException(person, "patient");
+		PersonValidator personValidator = new PersonValidator();
+		
+		try {
+			personValidator.validate(person, errors);
+		}catch(NullPointerException np) {
+			Assert.assertNotNull(np);
```

Comment:
@dkayiwa I reverted the old state. I would suggest to create a new ticket to fix this behaviour if you agree that this is not intended.

Reply:
Feel free to create a the new ticket.

Reply:
@dkayiwa are there any more issues now stopping the merging? As far as I am know all your comments are resolved now :)